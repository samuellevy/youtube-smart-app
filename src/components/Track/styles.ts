import styled from 'styled-components';

export const Container = styled.div`
  overflow: hidden;
  height: 80vh;
`;
export const Content = styled.div`
  display: flex;
`;
export const CategoryTitle = styled.h2``;
