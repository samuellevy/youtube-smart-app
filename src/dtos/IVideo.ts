export interface IVideo {
  id: string;
  publishedAt: string;
  channelId: string;
  title: string;
  description: string;
  thumbnail: {
    default: {url:string},
    medium: {url: string;}
    high: {url:string}
  };
  channelTitle: string;
  viewCount: number;
}
