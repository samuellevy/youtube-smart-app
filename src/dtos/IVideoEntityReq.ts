export interface IVideoEntityReq {
  id: string,
  snippet: IVideoSnippet,
  statistics: {
    viewCount: number;
  }
}

interface IVideoSnippet {
  publishedAt: string;
  channelId: string;
  channelTitle: string;
  title: string;
  description: string;
  thumbnails: IVideoThumbnail;
}

interface IVideoThumbnail {
  high: {
    url: string;
  }
}
