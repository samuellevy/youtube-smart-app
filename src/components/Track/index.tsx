import React, { useEffect, useState } from 'react';
import { useKeyboardContext } from '../../context/KeyboardContext';
import VideoCard from '../VideoCard';

import * as S from './styles';

interface ITrackComponent {
  active?:boolean;
  title?:string;
  id?:number;
}

interface IVideoCardItem {
  title: string;
  author: string;
  views: string;
  publishDate: string;
  active?: boolean;
}

const Track: React.FC<ITrackComponent> = ({ active, title, id }:ITrackComponent) => {
  const data = [
    {
      title: 'Harry Potter e a Pedra Filosofal',
      author: 'Warner Bros.',
      views: '1.3M',
      publishDate: '1 year ago',
    },
    {
      title: 'Harry Potter e a CÃ¢mara Secreta',
      author: 'Warner Bros.',
      views: '2.3M',
      publishDate: '2 years ago',
    },
  ];

  const [activeModule, setActiveModule] = useState(false);
  const [activeItem, setActiveItem] = useState(0);

  const { keyboard, dispatch }: any = useKeyboardContext();

  useEffect(() => {
    if (active === false) {
      setActiveItem(-1);
    } else {
      setActiveItem(0);
    }
  }, [active]);

  useEffect(() => {
    if (keyboard.component === 'home') {
      console.log(keyboard);
    }
  }, [keyboard]);

  return (
    <S.Container>
      <S.CategoryTitle>{title}</S.CategoryTitle>
      <S.Content>
        {data.map((item, key) => (
          <VideoCard data={item} active={activeItem === key} />
        ))}
      </S.Content>
    </S.Container>
  );
};

export default Track;
