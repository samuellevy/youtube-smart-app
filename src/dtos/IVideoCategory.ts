import { IVideo } from './IVideo';

export interface IVideoCategory {
  id: string;
  title: string;
  videos: IVideo[];
}
