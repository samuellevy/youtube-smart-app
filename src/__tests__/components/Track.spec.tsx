import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import { fireEvent, waitFor } from '@testing-library/react';
import Track from '../../components/Track';
import { IVideo } from '../../dtos/IVideo';

Enzyme.configure({ adapter: new Adapter() });

const mockedHandleOut = jest.fn();
const mockedHandlePlay = jest.fn();
const mockedKeyboard = jest.fn();
const mockedDispatch = jest.fn();
const mockVideoItem: IVideo[] = [{
  channelId: 'a', channelTitle: 'teste', description: 'description', id: 'id', publishedAt: 'published', thumbnail: { default: { url: '' }, high: { url: '' }, medium: { url: '' } }, title: 'title', viewCount: 1,
}];
const mockItem = { id: 'teste', title: 'Teste', videos: mockVideoItem };

jest.mock('../../context/KeyboardContext', () => ({
  useKeyboardContext: () => ({
    keyboard: () => mockedKeyboard(),
    dispatch: () => mockedDispatch(),
  }),
}));

describe('<Search /> component', () => {
  const component = renderer.create(<Track
    handleOut={() => mockedHandleOut}
    handlePlay={() => jest.fn()}
    item={mockItem}
  />);

  const jsonComponent: ReactTestRendererJSON | any = component.toJSON();

  it('should render properly', () => {
    expect(jsonComponent).toMatchSnapshot();
  });

  it('should render container div component', () => {
    const tree = shallow(<Track
      handleOut={() => mockedHandleOut}
      handlePlay={() => mockedHandlePlay}
      item={mockItem}
    />).dive();

    expect(tree.exists('div')).toBeTruthy();
  });

  it('should render one Track', () => {
    expect(jsonComponent.children).toHaveLength(1);
  });
});
