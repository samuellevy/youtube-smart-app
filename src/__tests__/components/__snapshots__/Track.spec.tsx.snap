// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> component should render properly 1`] = `
<div
  className="sc-iBPRYJ dUyfnD"
>
  <h1>
    Carregando...
  </h1>
</div>
`;
